modules/ROOT/pages/dataweave-cookbook-flatten-arrays.adoc

```diff
@@ -0,0 +1,316 @@
+= Flatten Elements of Arrays
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+:keywords: studio, anypoint, transform, transformer, format, aggregate, rename, split, filter convert, xml, json, csv, pojo, java object, metadata, dataweave, data weave, datamapper, dwl, dfl, dw, output structure, input structure, map, mapping
+
+DataWeave can flatten subarrays of an array and collections of key-value pairs
+within DataWeave objects, arrays, and subarrays.
+//LINK TO DW 1.0 EXAMPLES:
+include::partial$dataweave1-links.adoc[tag=dataweave1Examples]
+
+=== Flatten Subarrays with Key-Value Pairs into an Array of Objects
+
+This example shows how `flatten` acts on key-value pairs of the input array
+defined by the variable `arrayOne`. Notice that every key-value pair in the
+array becomes a separate DataWeave object.
+
+The example uses this function:
+
+* `flatten` to move the elements from the subarrays to the parent array,
+eliminate the subarrays, and covert all key-value pairs into a list of objects
+within the parent array.
+
+.DataWeave:
+[source,dataweave,linenums]
+----
+%dw 2.0
+var arrayOne = [
+                 [
+                    "keyOne" : 1,
+                    "keyTwo" : 2
+                 ],
+                 [
+                    "keyThree" : 3,
+                    "keyFour" : 4,
+                    "keyFive" : 5
+                 ],
+                 "keySix" : 6
+               ]
+output application/json
+---
+flatten(arrayOne)
+----
+
+.Output JSON:
+[source,json,linenums]
+----
+[
+  {
+    "keyOne": 1
+  },
+  {
+    "keyTwo": 2
+  },
+  {
+    "keyThree": 3
+  },
+  {
+    "keyFour": 4
+  },
+  {
+    "keyFive": 5
+  },
+  {
+    "keySix": 6
+  }
+]
+----
+
+=== Flatten Combined Arrays
+
+Both DataWeave examples return an array that combines and flattens the
```

Comment:
```suggestion
Like the previous DataWeave example, the following returns an array that combines and flattens the
```

Reply:
As I get further into this topic set, I become confused about "both examples" in light of the fact that there are actually many examples. Do we mean "both" as in the two subtopics?
GLOBALLY search for "example" and evaluate whether the context makes clear what we mean when we refer to previous or next or first or second, and so on.